### DISCUSSION TITLE
New open street map importer generates too narrow streets.

### DISCUSSION THREAD

**Discussion by drblallo** [carla-simulator/carla#7465] [Q&A]
Hi, 
i have been using the procedural generation tool released with carla 0.9.15 for a couple of days. While the process to import a map has been smooth, the results generate too narrow roads that cannot be navigated by cars, and artifacts are generated as well.

For example, by importing the open street map section at position [N:45.48433, E:9.23697, S:45.47888, W:9.22467] , i get a incorrect street. The screenshot is just one of many streets that are blocked off from their natural continuation. 
![Screenshot_2024-04-18_15-57-55](https://github.com/carla-simulator/carla/assets/19780809/da24139b-8ee2-48a2-8998-54c93d0a12a7)

I tried to investigate why this happens, openstreetmap description of that location, seems to be perfectly correct, i did not managed to figure out what mechanism within calra ends up generating this.

Furthermore, streets end up generated too narrow. I am not sure why this happens, i think it is because openstreet map does not specify the width, and carla uses the default one.  I understand that without information carla can only guess, still i guess that the default road size should be exposed as a configurable parameter.

![Screenshot_2024-04-18_16-01-23](https://github.com/carla-simulator/carla/assets/19780809/6b1b691f-1ee3-4aa1-bc31-394a3f81ad5b)

Here two cars tried to turn left after leaving their original street, but the turn was too narrow and they got stuck 

Finally, cars end up disappearing randomly. I have not been able to figure out why.

In principle the first two problems are not a issue since i could edit the map by hand, the issue is that from what i understand the town is generated by a script at start time of the level, and i cannot fix the issues myself.

Is there something i missing about this? I understand that the state of this feature is experimental, but as it is right now i don't think it can be used for any purpose more complex than importing a single road.

**Comment by Blyron** [comment]
We are aware of this issues. This comes from osm2odr library; We have to continue developing and syncronizing with sumo in order to get a better translation from xodr to osm

**Comment by drblallo** [comment]
i see, thank you for your answer. Are improvements about this issues already planned? And in general, is there a set of improvements we may expect over the next 12 months? This may impact decisions about what features to rely on in our projects.

In principle, if fixing some of these issue is not too large of a task I would be willing to try fix some of them myself and then upstream them.

**Reply by Blyron** [reply]
We are using SUMO for translate osm to xodr 
https://github.com/carla-simulator/sumo/tree/aaron/defaultsidewalkwidth
this is the branch we are looking for.
We do not know when are we gonna improve it. We are making trasition to UE5 which is our main focus.
